### PR TITLE
Search Form : Allow to override Index used

### LINKED ISSUE #11164: Search Form : Allow to override Index used by using a QueryString/Form param
### Discussed in https://github.com/OrchardCMS/OrchardCore/discussions/11163

<div type='discussions-op-text'>

<sup>Originally posted by **saleemhaq** February 11, 2022</sup>
As i have conent type: Blog inside that there is alot of blog posts need to search inside of that content type of blog just, how can i limit full text search based on that content type?

As i see post full text search here:
https://docs.orchardcore.net/en/dev/docs/guides/implement-fulltext-search/ </div>

### PR DESCRIPTION
**Pull request by Skrypt**
Fixes #11164 

Allows passing a QueryString param in the search form page to override the current default Index used for displaying results. This can also be set as a hidden form input fixed value if used from a different form. These Indices are protected by permissions so it should be fine to allow to do this instead of needing to create a different controller per index required to be searched from.

Usage : /search?Terms=moon&Index=BlogIndex

### CODE REVIEW

**Comment by saleemhaq** [comment]
Guys who have write access so please merege that work: Pull Request here : https://github.com/OrchardCMS/OrchardCore/pull/11165 please.

**Comment by hishamco** [comment]
@saleemhaq please don't push the team, all the OC dev have permission, but every PR should be validated from different anglee, some may take hours and some may takes months to get them merge

I advice you fork the repo and apply these changes if you are in urgent case, otherwise be patient until the reviewer approve this

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs:80]: As mentioned by the below summary, here `searchSettings` should not be mutated, you can use a separate variable initialized with `searchSettings.SearchIndex` and that may be overridden by `viewModel.Index`, and then use it.

https://github.com/OrchardCMS/OrchardCore/blob/7c3d8019f77e6f27e87b754ae78f910216c017df/src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/ISiteService.cs#L15-L18

**Review by jtkech — APPROVED** [review]

**Pull request merged**
